core/shared/src/main/scala/zio/syntax/IOSyntax.scala

```diff
@@ -39,21 +39,23 @@ object IOSyntax {
     def collectAll: IO[E, List[A]]                     = IO.collectAll(ios)
   }
 
-  final class IOTuple2[E, A, B](val ios2: (IO[E, A], IO[E, B])) extends AnyVal {
-    def map2[C](f: (A, B) => C): IO[E, C] = ios2._1.flatMap(a => ios2._2.map(f(a, _)))
+  final class IOTuple2[E, RA, A, RB, B](val ios2: (ZIO[RA, E, A], ZIO[RB, E, B])) extends AnyVal {
```

Comment:
There is no need to use 2 different `R` type parameters: you can use one, and let Scala unify them. In fact, letting Scala do this initially instead of trying to force it later using `with` will lead to better inference.

On both this and the other tuple syntaxes, I'd go down to 1 `R` and let Scala do the heavy lifting...

Reply:
Wow, you're right @jdegoes – I'm not sure why I'm surprised that the inference works, but it works just fine that way and is much simpler.  Updated the PR (if you can still call it that 😄)